features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplayConfiguration.kt

```diff
@@ -35,7 +36,9 @@ data class SessionReplayConfiguration internal constructor(
      * means no session will be recorded, 100 means all sessions will be recorded.
      * If this value is not provided then Session Replay will default to a 100 sample rate.
      */
-    class Builder(@FloatRange(from = 0.0, to = 100.0) private val sampleRate: Float = SAMPLE_IN_ALL_SESSIONS) {
+    @Suppress("TooManyFunctions")
+    class Builder(@FloatRange(from = 0.0, to = 100.0) private val sampleRate: Float = SAMPLE_IN_ALL_SESSIONS,
+                  private val logger: InternalLogger = InternalLogger.UNBOUND) {
```

Comment:
let's maybe split constructor of this Builder into 2: one public which takes `sampleRate` only and another one is internal which has `InternalLogger` as a parameter? The reason for doing that is that `InternalLogger` is not meant to be used by the end-user, so we shouldn't expose an opportunity to inject it, leaving end-user puzzled about the options.

Reply:
I've split it and created an internal constructor